### PR TITLE
feat(MultiIndex): remove the need for virtual hits when using connectAutoComplete

### PR DESCRIPTION
**Pull request by mthuret**
see #15.

Now you can use `connectAutoComplete` like:

```js
<InstantSearch
      appId=""
      apiKey=""
      indexName="ikea"
    >
      <Configure hitsPerPage={1} />
      <Index indexName="bestbuy" />
      <Index indexName="airbnb" />
      <AutoComplete />
</InstantSearch>
```

### CODE REVIEW

File: packages/react-instantsearch/src/core/Index.js [hunk 1]
```diff
@@ -58,8 +84,11 @@ Index.propTypes = {
 };
 
 Index.childContextTypes = {
-  // @TODO: more precise widgets manager propType
```
Comment (Haroenv) [inline packages/react-instantsearch/src/core/Index.js]: Why was this todo removed, is it solved?

File: packages/react-instantsearch/src/core/Index.test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  it('exposes multi index context', () => {
+    const wrapper = mount(
+      <Index {...DEFAULT_PROPS}>
+        <div />
+      </Index>,
+      { context }
+    );
+
+    const childContext = wrapper.instance().getChildContext();
+    expect(childContext.multiIndexContext.targetedIndex).toBe('foobar');
```
Comment (Haroenv) [inline packages/react-instantsearch/src/core/Index.test.js]: `toBe(DEFAULT_PROPS.indexName)`

**Comment by vvo** [comment]
There's no example of connectAutoComplete in your PR description, am I missing smg?

**Comment by mthuret** [comment]
@vvo What do you mean? You have the recipe (link in the guide) and the story?

File: packages/react-instantsearch/examples/autocomplete/src/App-Multi-Index.js [hunk 3]
```diff
@@ -15,17 +12,11 @@ const App = () => (
   >
     <AutoComplete />
     <Configure hitsPerPage={1} />
-    <Index indexName="bestbuy">
-      <VirtualHits />
-    </Index>
-    <Index indexName="airbnb">
-      <VirtualHits />
```
Comment (vvo) [inline packages/react-instantsearch/examples/autocomplete/src/App-Multi-Index.js]: 👍

**Comment by vvo** [comment]
LGTM, does it have any impact on other usages?

**Comment by mthuret** [comment]
@vvo I don't think so. If you don't need the `connectAutoComplete` then you still use `<Index/>` as usual. The only thing now, is that if you're using it completely alone in that use case (but why?), then a request will be fired (right now it was fired if a least one widget was nested).

**Pull request merged**